website/docs/r/rde_interface_behavior.html.markdown

```diff
@@ -0,0 +1,101 @@
+---
+layout: "vcd"
+page_title: "VMware Cloud Director: vcd_rde_interface_behavior"
+sidebar_current: "docs-vcd-resource-rde-interface-behavior"
+description: |-
+   Provides the capability of managing RDE Interface Behaviors in VMware Cloud Director.
+---
+
+# vcd\_rde\_interface\_behavior
+
+Provides the capability of managing RDE Interface Behaviors in VMware Cloud Director.
+
+Supported in provider *v3.10+*. Requires System administrator privileges.
+
+~> Be aware that Behaviors can only be created and deleted when no [RDE Types](/providers/vmware/vcd/latest/docs/resources/rde_type) are using the Interface where they are defined.
+If you want to use RDE Types with Behaviors, you should use `depends_on` as seen in the example [here](/providers/vmware/vcd/latest/docs/resources/rde_interface_behavior#example-usage)
+
+## Example Usage
+
+```hcl
+resource "vcd_rde_interface" "my_interface" {
+  vendor  = "bigcorp"
+  nss     = "tech"
+  version = "1.2.3"
+  name    = "BigCorp Interface"
+}
+
+resource "vcd_rde_interface_behavior" "my_behavior" {
+  interface_id = vcd_rde_interface.my_interface.id
+  name         = "MyBehavior"
+  description  = "Adds a node to the cluster.\nParameters:\n  clusterId: the ID of the cluster\n  node: The node address\n"
+  execution = {
+    "id" : "MyExecution"
+    "type" : "Activity"
+  }
+}
+
+resource "vcd_rde_interface_behavior" "my_behavior2" {
+  interface_id = vcd_rde_interface.my_interface.id
+  name         = "MyBehavior2"
+  execution = {
+    "id" : "MyExecution2"
+    "type" : "noop"
+  }
+}
+
+resource "vcd_rde_interface_behavior" "my_behavior3" {
+  interface_id = vcd_rde_interface.my_interface.id
+  name         = "MyBehavior3"
+  execution = {
+    "type": "WebHook",
+    "id": "testWebHook",
+    "href": "https://hooks.slack.com:443/services/T07UZFN0N/B01EW5NC42D/rfjhHCGIwzuzQFrpPZiuLkIX" ,
+    "_internal_key": "secretKey"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `interface_id` - (Required) The ID of the RDE Interface that owns the Behavior
+* `name` - (Required) Name of the Behavior
+* `description` - (Optional) A description specifying the contract of the Behavior
+* `execution` - (Required) A map that specifies the Behavior execution mechanism.
+  You can find more information about the different execution types, like `WebHook`, `noop`, `Activity`, `MQTT`, `VRO`, `AWSLambdaFaaS`
+  and others [in the Extensibility SDK documentation](https://vmware.github.io/vcd-ext-sdk/docs/defined_entities_api/behaviors)
+
+## Attribute Reference
+
+* `ref` - The Behavior invocation reference to be used for polymorphic behavior invocations
```

Comment:
Would this actually mean that this `ref` can be used with `vcd_rde_type`?

Reply:
For Interfaces I believe that `ref` and `ID` will be always the same. This field probably makes more sense in the API as one can do more of an "swap" easier, between Interface and Type behaviors.